### ISSUE TITLE
Perfect for my 840 G5

### ISSUE THREAD

**Issue by crylessengine** [chris1111/HP-EliteBook-840-G4#2]
Dear Chris,

I just wanted to express my sincere appreciation for your work put into this amazing repo. It has been an invaluable resource for me. With a few small adjustments, I was able to get my 840 G5 working perfectly, even with the Kaby Lake refresh and 4k internal display. It even pairs seamlessly with my USB-C dock.

I was unable to achieve such success with any of the EFIs specific to my model, even when following the Dortania guide to build from scratch. Your repository has been a game-changer and has helped me greatly. While my camera is not yet working, it is a minor inconvenience compared to the incredible progress I've made thanks to your efforts.

Thank you so much for sharing your work with the community. Your contribution is greatly appreciated.

Let me know should you want more details on the hardware or changes I've made for myself. I'm not a professional developer, so I'm not planning to fork the repo.

Sincerely,

Miro

**Comment by chris1111** [comment]
Thanks i appreciate it and glad to see it works for you and your G5
Yes I am curious about your change

**Comment by crylessengine** [comment]
Changes I did for my G5:

1. Replaced the problematic NVME Samsung SM981. It was booting into installation, but crashing/freezing randomly. I only had Samsung SM961 from a broken laptop lying around, but that one has worked.
2. Wifi I had to use AirportItlwm kext. 
3. Battery indication didn’t work, so I dropped SSDT-BATT and added ECEnabler kext instead. I also had to clean up plist hacks under ACPI/Patch. I just googled what each does and removed those related to battery.
4. I fixed trackpad issues by redoing all Voodoo kext and config using Dortania guide.
5. Upgraded to OpenCore 0.8.8, replaced with latest SSDTs, used SSDT-EC-USBX-LAPTOP and SSDT-PLUG-DRTNIA as per guide. Also replaced all remaining kexts with latest versions. I believe that helped stabilize the whole setup.

Cheers!
